src/core/ext/filters/client_channel/lb_policy/priority/priority.cc

```diff
@@ -412,6 +414,8 @@ void PriorityLb::DeleteChild(ChildPriority* child) {
 }
 
 void PriorityLb::TryNextPriorityLocked(bool report_connecting) {
+  current_priority_ = UINT32_MAX;
+  current_child_from_before_update_ = nullptr;
```

Comment:
Moving these up because otherwises the current_prioirity_ could be retained as this method returns, preventing us from doing a SelectPrioirtyLocked in HandleChildConnectivityStateChangeLocked